webui/src/js/viewModels/model-design-view.js

```diff
@@ -3,81 +3,241 @@
  * Copyright (c) 2021, 2022, Oracle and/or its affiliates.
  * Licensed under The Universal Permissive License (UPL), Version 1.0 as shown at https://oss.oracle.com/licenses/upl/
  */
-define(['accUtils', 'utils/i18n', 'knockout', 'models/wkt-project', 'utils/url-catalog', 'utils/wkt-logger',
-  'ojs/ojinputtext', 'ojs/ojlabel', 'ojs/ojbutton', 'ojs/ojformlayout' ],
-function(accUtils, i18n, ko, project, urlCatalog) {
-  function ModelDesignViewModel() {
 
-    this.connected = () => {
-      accUtils.announce('Model design view loaded.', 'assertive');
-      // Implement further logic if needed
-    };
+define(['accUtils', 'utils/i18n', 'knockout', 'models/wkt-project', 'utils/url-catalog', 'utils/wkt-logger', 'wrc-frontend/core/parsers/yaml', 'wrc-frontend/integration/viewModels/utils',
+        'wdt-model-designer/loader', 'ojs/ojinputtext', 'ojs/ojlabel', 'ojs/ojbutton', 'ojs/ojformlayout'],
+  function(accUtils, i18n, ko, project, urlCatalog, wktLogger, YamlParser, ViewModelUtils) {
+    function ModelDesignViewModel() {
+      const self = this;
 
-    this.labelMapper = (labelId, payload) => {
-      return i18n.t(`model-design-${labelId}`, payload);
-    };
+      this.project = project;
+      this.designer;
+      this.dataProvider = {};
 
-    this.isLinux = () => {
-      return window.api.process.isLinux();
-    };
+      this.connected = function () {
+        accUtils.announce('Model design view loaded.', 'assertive');
+        // Set up a change subscription on wlRemoteConsolePort observable,
+        // which will primarily be used in the "WRC Not Installed Yet" and
+        // "WRC Process Was Killed or Died" use cases.
+        this.wlRemoteConsolePortSubscription = project.wdtModel.internal.wlRemoteConsolePort.subscribe((newValue) => {
+          wktLogger.debug('connected: newValue=%s', newValue); 
+          showWdtModelDesigner(newValue);
+        });
+        // Get WRC backend port, asynchronously
+        window.api.ipc.invoke('get-wrc-port')
+          .then(backendPort => {
+            if (backendPort !== project.wdtModel.internal.wlRemoteConsolePort()) {
```

Comment:
This is where the race condition I was speaking about earlier exists and you are not handling it (i.e., what if `backendPort` comes back as `undefined`)?  

Frankly, I would just delete all of this code after line 25 to the end of the function.